### PR TITLE
Feature dsl experiment generation with number of users parameter

### PR DESCRIPTION
**Pull request by jeppe-style**

### CODE REVIEW

**Comment by VincenzoFerme** [comment]
@jeppe-style fix https://www.codacy.com/app/BenchFlow/benchflow/pullRequest?prid=612662, or @igormoreno improve the settings if these warnings are ok to have. Thanks!

**Comment by jeppe-style** [comment]
Yes, I have changes that will solve these issues but I thought to do it in a second pull-request, basically I will remove the Try and throw an exception instead since that is easier to handle in Java anyway.

**Comment by jeppe-style** [comment]
@VincenzoFerme the code is now updated so that the checks pass and now it will be easier to use in Java.

@igormoreno  Some of the issues that are shown in codacy I don't get when running `mvn verify`.

File: benchflow-dsl/pom.xml [hunk 1]
```diff
@@ -36,13 +38,13 @@
 
         <dependency>
             <groupId>org.scala-lang.modules</groupId>
-            <artifactId>scala-xml_2.11</artifactId>
+            <artifactId>scala-xml_2.12</artifactId>
```
Comment (VincenzoFerme) [inline benchflow-dsl/pom.xml]: `2.12` can also be defined as properties, or probably can be even extracted from `<scala.xml.version>` by filtering out the last `.1`. This applies to all the cases in this, and other files.

File: benchflow-dsl/src/test/resources/tests/exploration/users/benchflow-test.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  goal:
+    type: load
+
+    exploration_space:
+      workload:
+        users:
+          values: [5, 10, 15, 20]
+
+
+  users: 10
```
Comment (VincenzoFerme) [inline benchflow-dsl/src/test/resources/tests/exploration/users/benchflow-test.yml]: Why do you have the setting also here, if you define an exploration space on users?

**Review by VincenzoFerme — CHANGES_REQUESTED** [review]
This is just a very quick review, we should plan more in depth reviews in next iterations. An issue has been opened to schedule a complete code review (#244)

**Review by igormoreno — APPROVED** [review]
- Consider using string interpolation if you need to do 
too many string concatenations. I committed a small example.

- It would be nice to try to get read of those hardcoded strings: "configuration.goal.exploration_space.workload.users.". Finding your way somehow to build them dynamically. But it looks like that's not something trivial. So I would leave it for another iteration

**Comment by igormoreno** [comment]
Issue regarding hardcoded strings: #249

**Comment by jeppe-style** [comment]
Thanks @igormoreno and @VincenzoFerme for your comments! I have now included/fixed them so we should be good to go.

**Review by VincenzoFerme — APPROVED** [review]

**Pull request merged**